sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -618,4 +620,43 @@ export class ApplicationStudentsController extends BaseController {
       }
     }
   }
+
+  /**
+   * Get application to request appeal.
+   ** Application eligible to be requested for
+   ** a change will be returned.
+   * @param applicationNumber
+   * @param userToken
+   * @returns application
+   */
+  @ApiOkResponse({
+    description: "Returns application which can be requested for change.",
+  })
+  @ApiNotFoundResponse({
+    description:
+      "Application either not found or not eligible to request for change.",
+  })
+  @Get(":applicationNumber/appeal")
+  async getApplicationToRequestChange(
+    @Param("applicationNumber") applicationNumber: string,
+    @UserToken() userToken: IUserToken,
+  ): Promise<ApplicationPrimaryDTO> {
+    const application =
+      await this.applicationService.getApplicationToRequestChange(
+        applicationNumber,
+        userToken.userId,
+      );
+    if (!application) {
+      throw new NotFoundException(
+        new ApiProcessError(
+          "Given application either does not exist or is not complete to request change.",
+          INVALID_APPLICATION_NUMBER,
+        ),
+      );
+    }
+    return {
+      id: application.id,
+      applicationNumber: application.applicationNumber,
+    };
```

Comment:
Can we add the cast to the return type `as ApplicationPrimaryDTO`?

Reply:
We had the same conversation in the past and, if I am not wrong, it was actually between you and @ann-aot when we talked about adding the cast in the return of the controller. Does it ring any bell?
I do not get where this fits into my comment: "Remember when we spoke once about assigning a dto instance to service method where service expects a service model it automatically casted"